src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java

```diff
@@ -28,8 +28,135 @@ public static void performCompatibilityUpdate() {
             } else {
                 newGen = genFields;
             }
-            // pr(newGen);
+
             Globals.prefs.put(JabRefPreferences.GENERAL_FIELDS, newGen);
         }
     }
+
+    /**
+     * Added from Jabref 2.11 beta 4 onwards to fix wrong encoding names
+     */
+    public static void upgradeFaultyEncodingStrings(){
+        JabRefPreferences prefs = Globals.prefs;
+        String defaultEncoding = prefs.get(prefs.DEFAULT_ENCODING);
+
+        if(defaultEncoding == null){
+            return;
+        }
+
+        switch(defaultEncoding){
```

Comment:
Maybe putting the upgrade mapping in a Map<String, String> would make the code a little bit more readable as the upgrade mapping is clearer to see.